storer/storer.go

```diff
@@ -0,0 +1,11 @@
+package storer
+
+import "encoding/json"
+
+// Storer writes data collected from Collector to some form
+// of persistent storage
+type Storer interface {
+	initialize() error
+	saveEntry(json json.Encoder, tag string) error
+	saveEntries(json json.Encoder, tag string) error
```

Comment:
If both are encoding json, do we really need both plural and singular?

Reply:
In my mind, yes. Reason being that for files, you will save each of those to a separate file whereas for Mongo it probably matters less

Reply:
I guess you're right that I don't necessarily see a case where you're really only saving one entry. I can remove the singular and we can just make helper functions if we think we need to

Reply:
Yeah I like that. Have the singular be a helper. Or change the plural to an array of json, cause then we get the type checking. This way if someone calls `saveEntries` with a singular Val we'll have to handle the case manually in the function. Or it becomes a runtime error.